### PR TITLE
Support #5783. UI: Test Replacement Table Import

### PR DESCRIPTION
**Pull request by Sadredneck**

### CODE REVIEW

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    private void sendFilesOnAdminImportBackup(boolean replaceExisting, boolean autoEnable, String... files) {
+        WebElement putTable = driver.findElement(By.id("putTable"));
+        WebElement inputFile = putTable.findElement(By.xpath(".//input[@type='file']"));
+        WebElement submitButton = driver.findElement(By.xpath(".//input[@type='submit']"));
+
+        setCheckbox(putTable.findElement(By.id("replaceExisting")), replaceExisting);
+        setCheckbox(putTable.findElement(By.id("autoEnable")), autoEnable);
+
+        ((JavascriptExecutor) driver).executeScript("arguments[0].removeAttribute('multiple')", inputFile);
```
Comment (VeselovAlex) [inline atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Write a comment describing this workaround for PhantomJS. Make it PhantomJS-specific. Move to separate method.

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            if (!webElement.isSelected())
+                webElement.click();
+        } else if (webElement.isSelected())
+            webElement.click();
+    }
+
+    private boolean checkTable(WebElement table, String[][] expectedResult) {
+        List<WebElement> findElements = table.findElements(By.xpath("./tbody/tr"));
+        if (findElements.size() != expectedResult.length)
+            return false;
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Fixed

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        driver.findElement(By.xpath("//*/input[@type='submit' and @value='Delete']")).click();
+
+        WebDriverWait wait = new WebDriverWait(driver, 1);
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@id=\"confirm-modal\"]/div/button[contains(text(), 'Yes')]")));
+
+        driver.findElement(By.xpath("//*[@id='confirm-modal']/div/button[contains(text(), 'Yes')]")).click();
+    }
+
+    private void setCheckbox(WebElement webElement, boolean on) {
+        if (on) {
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Fixed. Thanks for idea

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+    private void goToReplacementTablesPage() {
+        driver.findElement(By.xpath("//*/a/span[contains(text(),'Data')]")).click();
+        driver.findElement(By.xpath("//*/a[contains(text(),'Replacement Tables')]")).click();
+        Assert.assertEquals("Wrong page", driver.getCurrentUrl(), url + "/replacement-tables/");
+    }
+
+    private void goToAdminImportBackupPage() {
+        Actions action = new Actions(driver);
+        action.moveToElement(
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Refactored method

File: atsd_webtests/src/test/java/com/axibase/webtest/service/ReplacementTablesImportTest.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        driver.findElement(By.xpath("//*/input[@type='submit' and @value='Delete']")).click();
+
+        WebDriverWait wait = new WebDriverWait(driver, 1);
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@id=\"confirm-modal\"]/div/button[contains(text(), 'Yes')]")));
+
+        driver.findElement(By.xpath("//*[@id='confirm-modal']/div/button[contains(text(), 'Yes')]")).click();
+    }
+
+    private void setCheckbox(WebElement webElement, boolean on) {
+        if (on) {
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/ReplacementTablesImportTest.java]: fixed

File: atsd_webtests/src/test/java/com/axibase/webtest/service/ReplacementTablesImportTest.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                    !tdList.get(1).getText().equals(expectedResult[i][0]) &&
+                    !tdList.get(2).getText().equals(expectedResult[i][1]) &&
+                    !tdList.get(3).getText().equals(expectedResult[i][2]))
+                return false;
+        }
+
+        return true;
+    }
+
+    private String[][] expectedResult() {
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/ReplacementTablesImportTest.java]: Moved to variable array

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+        goToAdminImportBackupPage();
+
+        sendFilesOnAdminImportBackup(false, false, testFile);
+
+        goToReplacementTablesPage();
+
+        Assert.assertTrue("Wrong table content",
+                checkTable(driver.findElement(By.id("overviewTable")), expectedResult()));
+
+        deleteReplacementTables();
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Added `@After` method

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    @Before
+    public void setUp() {
+        login();
+    }
+
+    @Test
+    public void testImportDataImportBackupPage() {
+        goToAdminImportBackupPage();
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Fixed

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                checkTable(driver.findElement(By.id("overviewTable")), expectedResult()));
+
+        deleteReplacementTables();
+    }
+
+    @Test
+    public void testImportDataImportBackupPageWithReplace() {
+        goToAdminImportBackupPage();
+
+        sendFilesOnAdminImportBackup(false, false, testFile);
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Created set methods for replacement

File: atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java (hunk 1 above)
Comment (Sadredneck) [reply atsd_webtests/src/test/java/com/axibase/webtest/service/AdminBackupImportTest.java]: Moved to method. Added comments

**Pull request merged**